### ISSUE TITLE
 A dictionary to translate change tasks to source code

### ISSUE THREAD

**Issue by WeiFoo** [opensciences/opensciences.github.io#173]
### Notes
__Categorization:__ Other
__Authors:__ [Katja Kevic](mailto:kevic@ifi.uzh.ch),  [Thomas Fritz](mailto:fritz@ifi.uzh.ch)
__General overview of the data:__

For  the empirical analysis, the authors created two datasets: dataset1 includes all change tasks that have at least one change set associated with them; dataset 2 is a subset of dataset 1 and includes all change tasks that have at least one change set and  one  task  context  associated.   For  the  AspectJ  project the authors only created dataset 1, since only two tasks had a task context associated. For each project the authors split each dataset into a training set to create the dictionary and a test set.  For  reasonable training  and  test  sets  and  to  be  able  to  compare  them  across projects, they chose the test set size to be ten. Also, to be able to compare the results of the dictionaries using dierent in-formation sources (change sets and/or task contexts) as well as preserving the chronological order, they chose the last ten change tasks in dataset 2 as the test set for Mylyn.Tasks,Mylyn. Context and Remus and the last ten change tasks in dataset 1 for AspectJ. All change tasks that were resolved before the ones in the test set were used as the training set.
__Link:__ [A dictionary to translate change tasks to source code](http://dl.acm.org/citation.cfm?id=2597095&CFID=626469285&CFTOKEN=55193119)
__Bibtex:__ 
```tex
@inproceedings{Kevic:2014:DTC:2597073.2597095,
 author = {Kevic, Katja and Fritz, Thomas},
 title = {A Dictionary to Translate Change Tasks to Source Code},
 booktitle = {Proceedings of the 11th Working Conference on Mining Software Repositories},
 series = {MSR 2014},
 year = {2014},
 isbn = {978-1-4503-2863-0},
 location = {Hyderabad, India},
 pages = {320--323},
 numpages = {4},
 url = {http://doi.acm.org/10.1145/2597073.2597095},
 doi = {10.1145/2597073.2597095},
 acmid = {2597095},
 publisher = {ACM},
 address = {New York, NY, USA},
 keywords = {Dictionary, change task, interaction history, location},
}
```
__Data Set:__ https://bitbucket.org/kkevic/datasets-nl-src-dictionary/downloads

**Comment by vivekaxl** [comment]
typo corrrected!

**Comment by CarterPape** [comment]
Needs description text

**Comment by reesjones** [comment]
Needs new context notes. See [this example](https://github.com/opensciences/opensciences.github.io/issues/156) or [this example](http://openscience.us/repo/issues/bugsal.html) as a reference.

Context notes are directly inserted into the actual repository (like [this](http://openscience.us/repo/issues/bugsal.html)), so good context notes are necessary for putting the datasets into the repo in a timely fashion.

**Comment by reesjones** [comment]
Added as taskdict.
http://openscience.us/repo/other/taskdict.html

**Comment by CarterPape** [comment]
Not good enough example
